torch/onnx/_internal/fx/passes/readability.py

```diff
@@ -0,0 +1,111 @@
+        nodes: Sequence[torch.fx.Node],
+        new_name: str,
+    ) -> None:
+        """Rename the parameter/buffer and replace corresponding nodes with new nodes of updated target.
+        """
+        assert len(nodes) > 0, "`nodes` cannot be empty"
+        assert (
+            len({node.target for node in nodes}) == 1
+        ), "`nodes` must all have same `target`"
+        old_name = nodes[0].target
+        assert isinstance(old_name, str), f"Expected str, got type({old_name})"
+        # Parameter/buffer name cannot contain "."
+        normalized_name = new_name.replace(".", "_")
+        attr_value = getattr(self.module, old_name)
+        setattr(self.module, normalized_name, attr_value)
+        delattr(self.module, old_name)
+        for node in nodes:
+            with self.module.graph.inserting_before(node):
+                new_node = self.module.graph.get_attr(normalized_name)
+                new_node.meta = node.meta
+                node.replace_all_uses_with(new_node)
+                self.module.graph.erase_node(node)
+        diagnostic.with_additional_message(
+            f"Renamed 'self.{old_name}' to 'self.{normalized_name}', "
+            f"normalized from original parameter name '{new_name}'."
+        )
+
+    def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
+        """Restore parameter and buffer names from original module.
+
+        For each `get_attr` node, if the target is a str representing a parameter or buffer
+        under `self.module`, we rename the parameter or buffer to its original name.
+        The parameters and buffers between `self.module` and `self.original_module` refer
+        to the same objects, allowing us to use it as key to retrieve the original name.
+        """
+        assert len(args) == 0, "RestoreParameterAndBufferNames does not take any args"
+        assert (
+            len(kwargs) == 0
+        ), "RestoreParameterAndBufferNames does not take any kwargs"
+        state_to_readable_name: Dict[Union[torch.nn.Parameter, torch.Tensor], str] = {}
+        state_to_readable_name.update(
+            {v: k for k, v in self.original_module.named_parameters()}
+        )
+        state_to_readable_name.update(
+            {v: k for k, v in self.original_module.named_buffers()}
+        )
+        diagnostic = self.diagnostic_context.inflight_diagnostic()
+
+        old_name_to_nodes: Dict[str, Tuple[List[torch.fx.Node], str]] = {}
+
+        for node in self.module.graph.nodes:
+            if node.op == "get_attr":
+                assert isinstance(
+                    node.target, str
+                ), f"Expected str, got type({node.target})"
+                if node.target in old_name_to_nodes:
+                    # We have already processed this parameter/buffer.
+                    old_name_to_nodes[node.target][0].append(node)
+                    continue
+                attr_value = getattr(self.module, node.target)
```

Comment:
Does it work with `node.target="self.h0.h1.h2.weight"`?

Reply:
Good question. No it does not. However, the example you give should never happen in our current code path. all parameters and buffers are top level after running aten decomp.

I'm putting a check in this PR. I'm not in favor of actually supporting this because 1. this means something unexpected happened at fx generation stage, 2. it adds a lot of unnecessary complexity. I'm open to discussion if you feel strongly otherwise.

Reply:
At least document in the pass docstring that a decomposition is assumed to be done before this pass. This creates a order dependency between the pass, but hopefully decomposition will be done by torch._dynamo.expor4t in the future and this dependency will be gone

Reply:
Ok, added

    This pass must be run after `Decompose` pass. Because this pass is expected to be called on
    `fx.GraphModule` produced by `proxy_tensor.make_fx`, where all parameters and buffers
    are registered at root level.